src/VisualStudio/Core/Impl/Options/AbstractOptionPage.cs

```diff
@@ -50,16 +52,22 @@ protected override void OnActivate(System.ComponentModel.CancelEventArgs e)
         {
             EnsureOptionPageCreated();
 
-            if (s_needsLoadOnNextActivate)
+            if (s_needsToUpdateOptionStore)
             {
                 // Reset the option store to the current state of the options.
                 s_optionStore.SetOptions(s_optionService.GetOptions());
                 s_optionStore.SetRegisteredOptions(s_optionService.GetRegisteredOptions());
 
-                s_needsLoadOnNextActivate = false;
+                s_needsToUpdateOptionStore = false;
             }
 
-            pageControl.LoadSettings();
+            if (_needsLoadOnNextActivate)
+            {
+                // For pages that don't use option bindings we need to load setting changes.
+                pageControl.LoadSettings();
```

Comment:
Consider changing `LoadSettings` to _something_ like `LoadNonDataboundSettings`? `LoadSettingsThatDoNotUseOptionBinding`? ¯\(°_o)/¯